src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/JSON2SolutionGraphImpl.java

```diff
@@ -0,0 +1,233 @@
+                    parseJSON(value.getAsObject(),blankNodes,solutionGraph));
+            }
+            else if(value.isArray() && key.startsWith(config.getJsonScalarKeyPrefix())){
+                // Value is array of scalar values
+                for(final JsonValue arrayScalar : value.getAsArray()){
+                    if(arrayScalar.isNull()){
+                        continue;
+                    }
+                    currentBlankNode.addLiteral(
+                        solutionGraph.createProperty(config.mapFieldToProperty(graphqlType,graphqlFieldName)), 
+                        jsonPrimitiveToLiteral(arrayScalar, graphqlType, graphqlFieldName, solutionGraph));
+                    
+                }
+            }
+            else if(value.isPrimitive()){
+                // Value is a single scalar value
+                currentBlankNode.addLiteral(
+                    solutionGraph.createProperty(config.mapFieldToProperty(graphqlType,graphqlFieldName)), 
+                    jsonPrimitiveToLiteral(value, graphqlType, graphqlFieldName, solutionGraph));
+            }
+        }
+
+        return currentBlankNode;
+    }
+
+    /**
+     * Removes the prefix from the json key @param key
+     */
+    protected String removeJsonKeyPrefix(final String key){
+        final int splitIndex = key.indexOf("_") + 1;
+        return key.substring(splitIndex);
+    }
+
+    /**
+     * Utility function used to create a Literal from a json value @param primitive
+     * usable by @param model. @param graphqlTypeName and @param graphqlFieldName are 
+     * used to fetch value type data from the GraphQLEndpoint to ensure that the correct 
+     * XML datatype is used when creating the literal.
+     */
+    protected Literal jsonPrimitiveToLiteral(final JsonValue primitive, final String graphqlTypeName, 
+            final String graphqlFieldName, final Model model){
+
+        assert primitive.isPrimitive();
+
+        final String valueType = endpoint.getGraphQLFieldValueType(graphqlTypeName, graphqlFieldName);
+
+        // The scalar value types from GraphQL ( GraphQL String, ID and custom scalar types etc. are assumed as XSD Strings)
+        switch(valueType){
+            case "Int":
+            case "Int!":
+                return model.createTypedLiteral(primitive.getAsNumber().value(), XSDPlainType.XSDint);
+            case "Boolean":
+            case "Boolean!":
+                return model.createTypedLiteral(primitive.getAsBoolean().value(), XSDPlainType.XSDboolean);
+            case "Float":
+            case "Float!":
+                return model.createTypedLiteral(primitive.getAsNumber().value(), XSDPlainType.XSDdouble);
+            default:
+                return model.createTypedLiteral(primitive.getAsString().value(), XSDPlainType.XSDstring);
+        }
+    }
+
+    /**
+     * Utility class used to represent a double valued key for Maps etc.
+     */
+    protected class BlankNodeID {
```

Comment:
This class can be replaced for a HeFQUIN's `Pair<String, String>` or a Jena's `PairOfSameType<String>`, since it doesn't add any special functionality.